homeassistant/components/yeelight/light.py

```diff
@@ -86,20 +97,57 @@ def _wrap(self, *args, **kwargs):
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the Yeelight bulbs."""
+    from yeelight.enums import PowerMode
+
+    data_key = '{}_lights'.format(DATA_YEELIGHT)
+
     if not discovery_info:
         return
 
-    yeelight_data = hass.data[DATA_YEELIGHT]
-    ipaddr = discovery_info[CONF_HOST]
-    device = yeelight_data[CONF_DEVICES][ipaddr]
+    if data_key not in hass.data.keys():
```

Comment:
Drop `.keys()`. It's not needed. If we want to be fancy we can use `dict.setdefault`.

Reply:
Fixed